src/component/SignUp.tsx

```diff
@@ -69,8 +72,11 @@ export function SignUp() {
   };
 
   // These are only some of the popup errors handled
-  function signInWithGoogle() {
-    signInWithPopup(getAuth(), provider).catch((error) => {
+  async function signInWithGoogle() {
+    try {
+      await signInWithPopup(getAuth(), provider);
+      history.push("/");
```

Comment:
`history.replace()` might be a better fit here, pushing adds to the stack whereas if a user clicks back presumably we don't want to take them back to the sign in screen.